External/Plugins/ASCompletion/Model/ASFileParser.cs

```diff
@@ -1755,6 +1766,63 @@ private ASMetaData LookupMeta(ref string ba, ref int i)
             return md;
         }
 
+        private ASMetaData LookupHaxeMeta(ref string ba, ref int i)
+        {
+            int len = ba.Length;
+            int i0 = i;
+            int line0 = line;
+            int inString = 0;
+            int parCount = 0;
+            bool isComplex = false;
+            while (i < len)
+            {
+                char c = ba[i];
+                if (inString == 0)
+                {
+                    if (c == '"') inString = 1;
+                    else if (c == '\'') inString = 2;
+                    else if ("{;[".IndexOf(c) >= 0)
+                    {
+                        i = i0;
+                        line = line0;
+                        return null;
+                    }
+                    else if (c == '(') parCount++;
+                    else if (c == ')')
+                    {
+                        parCount--;
+                        isComplex = true;
+                        if (parCount <= 0) break;
+                    }
+                    else if (c <= 32 && parCount <= 0)
+                    {
+                        break;
+                    }
+                }
+                else if (c == 10 || c == 13)
+                {
+                    line++;
+                    if (c == 13 && i < len && ba[i + 1] == 10) i++;
+                }
+                else if (inString == 1 && c == '"') inString = 0;
+                else if (inString == 2 && c == '\'') inString = 0;
+                else if (inString > 0 && (c == 10 || c == 13)) inString = 0;
```

Comment:
I think they should be handled correctly, but sure, I'll make unit tests to be on the safe side.